docs/CONTRIBUTING.md

```diff
@@ -0,0 +1,14 @@
+# Pre-requisite
+
+ * __Nox__ can be installed following these instructions: https://nox.thea.codes/en/stable/
+
+# Developing
+
+ 1. Add any new source file (extension `.py`) to the list of files to be checked in `noxfile.py`
+ 2. Run `nox` on the root of the repo
+
+# Rules
+
+`nox` will run the following: 
+  * `black` to format the code: https://black.readthedocs.io/en/stable/
```

Comment:
I am not familiar with `nox`; will it not run black as per the `noxfile.py`?

Reply:
`nox` on it's own will only run the lint session due to:  `nox.options.sessions = ["lint"]`    
The black session has to be run seperately with `nox -rs black`

Reply:
This way we only run the linter in the workflow

Reply:
Ah I see. Should we include black in the session to be run?

Reply:
That would conflict with the linter session that gets run on PRs
The linter is setup to fail on PRs that haven't been formatted with black - if we include black in the PR workflow then it will 'pass' the linter but the formatted code won't make it on to master

Reply:
ah ok, I normally have githooks to handle that.

Reply:
Docu updated to inform dev how to run `black` formatter before `nox`